ext/nnstreamer/tensor_filter/tensor_filter_python3.cc

```diff
@@ -587,13 +589,19 @@ PYCore::run (const GstTensorMemory *input, GstTensorMemory *output)
         output[i].data = PyArray_DATA (output_array);
         Py_XINCREF (output_array);
         outputArrayMap.insert (std::make_pair (output[i].data, output_array));
+        newlyAddedItems.push_back (output[i].data);
       } else {
         ml_loge ("Output tensor type/size is not matched\n");
         res = -2;
         break;
       }
     }
-
+    if (res != 0) {
+      // Clean up only items added in this function call
```

Comment:
minor, let's keep line-comment rule.
`/* Clean up only items added in this function call */`